src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSource.java

```diff
@@ -40,39 +40,83 @@
     private static final String HADOOP_PART_PREFIX = "part-";
-    public static final int DEFAULT_SPLIT_SIZE = 10485760;
+    public static final int DEFAULT_SPLIT_SIZE = 10 * 1024 * 1024; // 10MB
 
     private transient final JavaSparkContext ctx;
     public ReadsSparkSource(JavaSparkContext ctx) {
         this.ctx = ctx;
     }
 
+    private static long getInputSizeInBytes(final JavaSparkContext ctx, final String filePath) {
+        // GCS case
+        if (BucketUtils.isCloudStorageUrl(filePath)) {
+            // TODO: implement directory listing
+            return -1; // undetermined - use default parallelism
+        }
+        // local file or HDFS case
+        try {
+            Path path = new Path(filePath);
+            FileSystem fs = path.getFileSystem(ctx.hadoopConfiguration());
+            FileStatus[] stati = fs.globStatus(path);
+            if (stati == null || stati.length == 0) {
+                return -1L;
+            }
+            long size = 0;
+            for (FileStatus status : stati) {
+                if (status.isDirectory()) {
+                    for (FileStatus st : fs.listStatus(status.getPath())) {
+                        size += st.getLen();
+                    }
+                } else {
+                    size += status.getLen();
+                }
+            }
+            return size;
+        } catch (IOException e) {
+            throw new UserException("Failed to determine total input size of " + filePath + "\n Caused by:" + e.getMessage(), e);
+        }
+    }
 
     /**
-     * Loads Reads using Hadoop-BAM. For local files, readFileName must have the fully-qualified path,
-     * i.e., file:///path/to/bam.bam.
-     * @param readFileName file to load
-     * @param referencePath Reference path or null if not available. Reference is required for CRAM files.
-     * @param intervals intervals of reads to include.
-     * @return RDD of (SAMRecord-backed) GATKReads from the file.
+     * Return the recommended parallelism for a pipeline processing the given input path. The parallelism is
+     * calculated by finding the total size (in bytes) of all the files under the given input path, then
+     * dividing by the default split size (10MB).
+     * @param ctx the Spark context
+     * @param filePath the path to load files from
+     * @param defaultParallelism the default parallelism to use if the size of the input cannot be determined
+     * @return the recommended parallelism, or <code>defaultParallelism</code> if the size of the input
+     * cannot be determined
      */
-    public JavaRDD<GATKRead> getParallelReads(final String readFileName, final String referencePath, final List<SimpleInterval> intervals) {
-        return getParallelReads(readFileName, referencePath, intervals, DEFAULT_SPLIT_SIZE);
+    public static int getRecommendedParallelism(final JavaSparkContext ctx, final String filePath, final int defaultParallelism) {
```

Comment:
I think this should also live in a `SparkUtils`-type class instead of `ReadSparkSource`, as it's not really specific to reads.